ci-operator/step-registry/cucushift/upgrade/run/cucushift-upgrade-run-commands.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# run upgrade
+function upgrade() {
+    cmd="oc adm upgrade --to-image=${TARGET_RELEASE} --force=true --allow-explicit-upgrade=true --allow-upgrade-with-warnings=true"
+    echo "Running command: ${cmd}"
+    eval "${cmd}"
+}
+
+function monitor_cvo() {
+    INTERVAL=600
+    CNT=12
+
+    while [ $((CNT)) -gt 0 ]; do
+        # or history
+        version_full=$(oc get clusterversion/version -o json)
+        image=$(echo "${version_full}" | jq -r '.status.history[0].image')
+        if [[ "${image}" != "${TARGET_RELEASE}" ]]; then
```

Comment:
This check is not enough. When upgrade is triggered, the image is updated. We need check state and version together.